File: core/src/main/java/org/ecos/reader/core/io/files/ZipFileImpl.java
package org.ecos.reader.core.io.files;

import org.ecos.reader.core.io.files.exceptions.DoNoExistsException;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.zip.ZipEntry;
import java.util.zip.ZipInputStream;

import reader.ecos.org.core.io.files.exceptions.UnknownError;

import static java.lang.String.format;

@SuppressWarnings("unused")
public class ZipFileImpl implements ZipFile {
    private File mFileToWorkWith;

    @Override
    public void prepareToUse(File file) {
        mFileToWorkWith = file;
    }

    @Override
    public boolean exists() {
        String filesPathAsString = getFilePathAsString();

        try {
            new FileInputStream(filesPathAsString);
        } catch (FileNotFoundException e) {
            return false;
        }
        return true;
    }

    private String getFilePathAsString() {
        return mFileToWorkWith.getPath().asString();
    }

    @Override
    public FileCollection getFiles() throws DoNoExistsException, UnknownError {
        String filePathAsString = getFilePathAsString();
        if(!exists())
            throw new DoNoExistsException(format("The file you want to access do not exists in the path you're using (%s)", filePathAsString));

        FileCollection collectionOfInnerFiles = new FileCollectionImpl();

        try(ZipInputStream stream = new ZipInputStream(new FileInputStream(filePathAsString)))
        {
            ZipEntry entry;
            while((entry = stream.getNextEntry())!=null)
            {
                collectionOfInnerFiles.add(FileImpl.from(PathImpl.from(entry.getName())));
            }
        } catch (IOException e) {
            e.printStackTrace();
            throw new UnknownError(format("Message: %s",e));
        }
        return collectionOfInnerFiles;
    }
}

File: core/src/main/kotlin/org/ecos/reader/core/io/files/ZipFileImpl.kt
package org.ecos.reader.core.io.files

import org.ecos.reader.core.io.files.exceptions.DoNoExistsException

import java.io.FileInputStream
import java.io.FileNotFoundException
import java.io.IOException
import java.util.zip.ZipEntry
import java.util.zip.ZipInputStream

import org.ecos.reader.core.io.files.exceptions.UnknownError

import java.lang.String.format
import org.ecos.reader.core.io.files.FileImpl.Companion

class ZipFileImpl : ZipFile {
    private var mFileToWorkWith: File? = null

    override fun prepareToUse(file: File) {
        mFileToWorkWith = file
    }

    override fun exists(): Boolean {
        val filesPathAsString = filePathAsString

        try {
            FileInputStream(filesPathAsString)
        } catch (e: FileNotFoundException) {
            return false
        }

        return true
    }

    private val filePathAsString: String
        get() = mFileToWorkWith!!.path.asString()

    override val files: FileCollection
        @Throws(DoNoExistsException::class, UnknownError::class)
        get() {
            val filePathAsString = filePathAsString
            if (!exists())
                throw DoNoExistsException(format("The file you want to access do not exists in the path you're using (%s)", filePathAsString))

            val collectionOfInnerFiles = FileCollectionImpl()

            try {
                ZipInputStream(FileInputStream(filePathAsString)).use { stream ->
                    while (true){
                        val entry = stream.nextEntry ?: break
                        collectionOfInnerFiles.add(FileImpl.from(PathImpl.from(entry.name)))
                    }
                }
            } catch (e: IOException) {
                e.printStackTrace()
                throw UnknownError(format("Message: %s", e))
            }

            return collectionOfInnerFiles
        }
}